File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        priority_queue <long long> pq; 
        for(int i = 0; i< nums.size(); i++)
        {
            pq.push(nums[i]);
        }long long ans = 0;
        while(k--)
        {
            ans += (pq.top());
            int m = pq.top();
            double m2 = pq.top();
            m= m/3;
            m2 = m2/3;
            if(m!=m2)
                m= m+1;
            pq.pop();
            
            pq.push(m);
        }return ans ;

        
    }
};

File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.cs
public class Solution {
    public long MaxKelements(int[] nums, int k) {
        PriorityQueue <int,int> pq  = new PriorityQueue<int,int>(Comparer<int>.Create((a, b) => b-a)); 
        for(int i = 0; i< nums.Length; i++)
        {
            pq.Enqueue(nums[i],nums[i]);
        }long  ans = 0;
        while(k-- > 0)
        {
            ans += (pq.Peek());
            Console.WriteLine(pq.Peek());
            int m = pq.Peek();
            double m2 = pq.Peek();
            m= m/3;
            m2 = m2/3;
            if(m!=m2)
                m= m+1;
            pq.Dequeue();
            
            pq.Enqueue(m,m);
        }return ans ;
        
        
    }
}